test/acceptance/objects/setup_test.go

```diff
@@ -85,6 +86,55 @@ func TestObjects_AsyncIndexing(t *testing.T) {
 	asyncTestObjects(t)
 }
 
+// Tests for allocChecker nil error on dynamic indexes during shard load
+func TestObjects_AsyncIndexing_LoadShard(t *testing.T) {
+	ctx := context.Background()
+	compose, err := docker.New().
+		WithWeaviate().
+		WithWeaviateEnv("ASYNC_INDEXING", "true").
+		WithWeaviateEnv("PERSISTENCE_MIN_MMAP_SIZE", "20MB").
```

Comment:
I think it would be better to:
- disable async indexing (so everything is imported right away and we don't need sleeps)
- Set `PERSISTENCE_MAX_REUSE_WAL_SIZE` to 0. Then Segments are flushed immediately after deactivating a tenant and importing one object would be enough